yt_dlp/extractor/wasdtv.py

```diff
@@ -0,0 +1,169 @@
+# coding: utf-8
+from __future__ import unicode_literals
+
+from .common import InfoExtractor
+from ..utils import (
+    ExtractorError,
+    int_or_none,
+    parse_iso8601,
+    urljoin,
+    try_get,
+)
+
+
+class WASDTVBaseIE(InfoExtractor):
+    _API_BASE = 'https://wasd.tv/api/'
+
+    def _fetch(self, path, video_id, description, query={}):
+        response = self._download_json(
+            urljoin(self._API_BASE, path),
+            video_id=video_id,
+            note=f'Downloading {description} metadata',
+            errnote=f'Unable to download {description} metadata',
+            query=query)
+        error = response.get('error')
+        if error:
+            raise ExtractorError(f'{self.IE_NAME} returned error: {error}', expected=True)
+        return response.get('result')
+
+    def _extract_thumbnails(self, thumbnails_dict):
+        if not thumbnails_dict:
+            return None
+        thumbnails = []
+        for index, thumbnail_size in enumerate(['small', 'medium', 'large']):
+            thumbnail_url = thumbnails_dict.get(thumbnail_size)
+            if not thumbnail_url:
+                continue
+            thumbnails.append({
+                'url': thumbnail_url,
+                'preference': index,
+            })
+        return thumbnails
+
+    def _real_extract(self, url):
+        container = self._get_container(url)
+        stream = try_get(container, lambda x: x['media_container_streams'][0])
```

Comment:
Not a amndatory change, but fyi, this can be simplified to:
```suggestion
        stream = traverse_obj(container, ('media_container_streams', 0))
```